angular/src/lib/drawer/drawer.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+export type drawerOpenFromDirection = 'right' | 'top' | 'bottom';
+export type drawerSize = 'small' | 'medium' | 'large';
+
+
+
+@Component({
+  selector: 'md-drawer',
+  template: `
+  <div
+    class="md-drawer"
+    [ngClass]="[
+      isOpen ? 'md-drawer__open': '',
+      'md-drawer__'+ openFrom,
+      'md-drawer--' + size
```

Comment:
should openFrom and size be conditional as well? 
[ngClass]="[
          openFrom ? 'md-drawer__' + this.openFrom : '',
           size ? 'md-drawer__' + this.size: '',
        ]"

Reply:
I suppose it could.. the openFrom and size have defaults so if they aren't set by the user these will never be empty.